src/main/java/ru/hogwarts/school/controller/FacultyController.java

```diff
@@ -47,11 +42,16 @@ public Faculty delete(@PathVariable Long id) {
         return facultyService.delete(id);
     }
 
-    @GetMapping("/find-by-color")
-    public List<Faculty> getFacultyByColor(@RequestParam String color) {
+    @GetMapping(params = "color")
+    public List<Faculty> getFacultyByColor(@RequestParam (required = false) String color) {
         return facultyService.getFacultyByColor(color);
     }
 
+    @GetMapping(params = "nameOrColor")
+    public List<Faculty> findByNameOrColor(@RequestParam (required = false) String nameOrColor) {
```

Comment:
```suggestion
    public List<Faculty> findByNameOrColor(@RequestParam String nameOrColor) {
```
зачем required = false?